pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -71,6 +75,42 @@ type Registry struct {
 	Prefix string `toml:"prefix"`
 }
 
+// PullSource consists of an Endpoint and a Reference. Note that the reference is
+// rewritten according to the registries prefix and the Endpoint's location.
+type PullSource struct {
+	Endpoint  Endpoint
+	Reference reference.Named
+}
+
+// PullSourcesFromReference returns a slice of PullSource's based on the passed
+// reference.
+func (r *Registry) PullSourcesFromReference(ref reference.Named) ([]PullSource, error) {
+	if r.MirrorByDigestOnly {
```

Comment:
I would rather prefer refactoring this to have the conditional only determine the list of endpoints, and to share the `rewriteReference` loop.